include/counter.h

```diff
@@ -39,11 +39,91 @@ extern "C" {
 
 /**@} */
 
+/**@defgroup COUNTER_ALARM_FLAGS Counter device capabilities
+ * @{ */
+
+/**
+ * @brief Counter alarm absolute value flag.
+ *
+ * Ticks relation to counter value. If set ticks are treated as absolute value,
+ * else it is relative to the counter reading performed during the call.
+ */
+#define COUNTER_ALARM_FLAG_ABSOLUTE BIT(0)
+
+/**
+ * @brief Flag used internally indicating that detection of too late alarm
+ *	  setting is enabled.
+ *
+ * Alarm can be set too late when context is interrupted and counter progresses
+ * beyond requested value.
+ *
+ * Detection of too late setting is vital since if it is not detected alarm
+ * is delayed by full period of the counter (up to 32 bits). Because of
+ * wrapping, it is impossible to distinguish alarm which is short in the past
+ * from alarm which is targeted to expire after full counter period. In order to
+ * detect too late setting, longest possible alarm is limited. When driver can
+ * assume that user will not request alarm which is too far in the future, it
+ * can detect too late setting.
+ *
+ * Example 1:
+ * - Counter top value: 1024
+ * - Current counter value: 20
+ * - Maximal possible alarm value (relative to now): 1000.
+ * - Requested absolute alarm: 23
+ *
+ * Context is interrupted when alarm is being set. Value 23 was set in
+ * peripheral register when counter was at 25. Driver reads counter after
+ * setting the alarm and calculates the difference. Alarm is suppose to expire
+ * 1022 ticks from now (23 - 25 & 1024) which exceeds 1000 limitation.
+ * Late setting is detected.
+ *
+ * Example 2:
+ * - Counter top value: 1024
+ * - Current counter value: 20
+ * - Maximal possible alarm value (relative to now): 1000.
+ * - Requested absolute alarm: 1020
+ *
+ * Context is not interrupted. Alarm is set to 1020. Counter value is 20. Alarm
+ * set correctly. If counter would increase to 21 then difference would be 999
+ * which is below maximal possible alarm value which is considered correct
+ * setting.
+ *
+ * Maximal possible alarm value (relative to now) is related to the longest
+ * potential preemption which can occur when setting the alarm. It may vary
+ * between instances of the driver and is application specific. If counter
+ * period is long (>minutes), it is recommended to set with reserve, e.g. if
+ * counter period is 512 seconds, maximal relative alarm value can be set to
```

Comment:
The description should take into account the newly introduced concept of the guard period. The nomenclature used is inconsistent. Here we talk about setting relative alarm but the API doesn't provide any means of doing so.